### PR TITLE
[SPARK-28677][WEBUI] "Select All" checkbox in StagePage doesn't work properly

### PR DESCRIPTION
**Pull request by sarutak**
## What changes were proposed in this pull request?

In StagePage, only the first optional column (Scheduler Delay, in this case) appears even though "Select All" checkbox is checked.

Screenshot from 2019-08-09 18-46-05

The cause is that wrong method is used to manipulate multiple columns. columns should have been used but column was used.
I've fixed this issue by replacing the `column` with `columns`.

## How was this patch tested?

Confirmed behavior of the check-box.

Screenshot from 2019-08-09 18-54-33

### CODE REVIEW

**Comment by sarutak** [comment]
CC: @pgandhi999 and @tgravescs

**Comment by sarutak** [comment]
retest this please.

**Comment by pgandhi999** [comment]
@sarutak Thank you for catching the bug. Interestingly, I am only able to reproduce the bug in Spark 3.0, it works fine in Spark 2.3 and 2.4. Are you also able to reproduce the bug in 2.3 and 2.4?

File: core/src/main/resources/org/apache/spark/ui/static/stagepage.js [hunk 1]
```diff
@@ -897,7 +897,7 @@ $(document).ready(function () {
                     // Get the column
                     var para = $(this).attr('data-column');
                     if (para == "0") {
-                        var column = taskTableSelector.column(optionalColumns);
+                        var column = taskTableSelector.columns(optionalColumns);
```
Comment (pgandhi999) [inline core/src/main/resources/org/apache/spark/ui/static/stagepage.js]: Could you rename the variable to something like `allColumns` to have more clarity?

**Review by srowen — APPROVED** [review]
OK by me if it's a clean UI fix.

**Review by pgandhi999 — CHANGES_REQUESTED** [review]

**Review by pgandhi999 — APPROVED** [review]

**Comment by sarutak** [comment]
@pgandhi999 
> Thank you for catching the bug. Interestingly, I am only able to reproduce the bug in Spark 3.0, it works fine in Spark 2.3 and 2.4. Are you also able to reproduce the bug in 2.3 and 2.4?

`2.3` and `2.4` use `additional-metrics.js` rather than `stagepage.js` for the checkboxes.
I think that's why only `3.0` has this issue.

**Comment by pgandhi999** [comment]
> 2.3 and 2.4 use additional-metrics.js rather than stagepage.js for the checkboxes.
I think that's why only 3.0 has this issue.

Ah, that makes sense. Thank you.

**Comment by pgandhi999** [comment]
LGTM +1

**Comment by tgravescs** [comment]
"2.3 and 2.4 use additional-metrics.js rather than stagepage.js for the checkboxes.
I think that's why only 3.0 has this issue."

Can you please explain this in more details so we have it documented what really book it?  Please also update the description to say what you change and ideally why it was broken.

**Comment by sarutak** [comment]
> Can you please explain this in more details so we have it documented what really book it? Please 

`stagepage.js` is not present in `2.3` and `2.4`. This bug is introduced by `stagepage.js` which is merged into `master`.
The cause is that wrong method is used to manipulate multiple columns. `columns` should have been used but `column` was used.

**Review by tgravescs — APPROVED** [review]
changes look good.  thanks @sarutak

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**